### PR TITLE
Various refactors + improvements to project dropdown in invite member modal

### PR DESCRIPTION
**Pull request by joshenlim**
- Remove unnecessary height classes to Select components
- Use FormItemLayout components
- Use ComboBox for project dropdown + ScrollArea

### CODE REVIEW

File: apps/studio/components/interfaces/Organization/TeamSettings/InviteMemberButton.tsx [hunk 1]
```diff
@@ -79,6 +87,8 @@ export const InviteMemberButton = () => {
     { enabled: canReadSubscriptions }
   )
   const currentPlan = subscription?.plan
+  const isOptedIntoProjectLevelPermissions =
+    useIsOptedIntoProjectLevelPermissions(slug as string) && currentPlan?.id === 'enterprise'
```
Comment (MildTomato) [inline apps/studio/components/interfaces/Organization/TeamSettings/InviteMemberButton.tsx:91]: this ok? @joshenlim

looks like logic change

File: packages/ui-patterns/form/Layout/FormLayout.tsx [hunk 2]
```diff
@@ -290,14 +290,14 @@ export const FormLayout = React.forwardRef<
         >
           {description}
         </FormDescription_Shadcn_>
-      ) : (
+      ) : description ? (
```
Comment (MildTomato) [inline packages/ui-patterns/form/Layout/FormLayout.tsx:293]: nice!

File: apps/studio/components/interfaces/Organization/TeamSettings/InviteMemberButton.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/Organization/TeamSettings/InviteMemberButton.tsx:91]: yup! this one's okay - a user should only be opted into project level permissions based on this 2 conditions
- whether is opted in via flag
- and if the plan is enterprise

**Review by alaister — APPROVED** [review]
LGTM!

- Tested opening the modal in an org with plp enabled

**Pull request merged**